core/src/main/java/com/orientechnologies/orient/core/sql/functions/coll/OSQLFunctionFirst.java

```diff
@@ -50,9 +50,7 @@ public Object execute(Object iThis, final OIdentifiable iCurrentRecord, Object i
     if (OMultiValue.isMultiValue(value))
       value = OMultiValue.getFirstValue(value);
 
-    if (first == this)
-      // ONLY THE FIRST TIME
-      first = value;
+    first = value;

```

Comment:
This line assign each value to the "first" variable. So it will have the last value.